service/books/dto.go

```diff
@@ -0,0 +1,22 @@
+package books
+
+type createBookRequest struct {
```

Comment:
Although in the exercise's specs  there is an alignment between the model and the request, that fact that its ties both the client and the DB (elastic) can create unhealthy coupling between the 2, being  more future-proof by separating the 2 is a common practice and can save a lot of pain in the future (Hypothetically speaking, as this is an excretes) 
So I'm quite against merging the request with the DB model.

Reply:
In our services, we mostly don't use request and response structs per route. This is not our code style in Go.
The alignment you say there is between the model and the request is not coincidental, this is a Create book request for the book model.
As you saw, when defining a struct in go we define the JSON value for each field.
Since the book is in JSON format both in ES and in the request there is no need for two strcuts and performing the conversion. You can use the same one for parsing the request and indexing to ES and vice versa. If the book definition will change, we only have to change the book model.
WDYT?
WDYT?